mteb/encoder_interface.py

```diff
@@ -151,3 +156,56 @@ def convert_conv_history_to_query(conversations: Sequence[Sequence[str]]) -> str
             The query.
         """
         ...
+
+
+class ImageEncoder:
+    """Interface for image encoder designed based on VLM2VecWrapper.
+    There is not a perfect 1-1 match, e.g. device can be None here.
+    The intention here is to define the current interface and adapt to as close to MTEB as possible
+    and align as much as possible with sentencetransformers.
+    """
+
+    def __init__(
+        self,
+        device: str | None,
+        **kwargs: Any,
+    ):
+        pass
+
+    def encode(  # current a 1-1 match with Encoder.encode
+        self,
+        sentences: Sequence[str],
+        *,
+        task_name: str,
+        prompt_type: PromptType | None = None,
+        **kwargs: Any,
+    ) -> np.ndarray:
+        pass
+
+    def get_image_embeddings( # Seems like sentence transformers use a singular encode for both images and text. Not sure if we want to do the same.
+        # If not it might be ideal to redefine Encoder.encode
+        self,
+        images: list[Image.Image] | DataLoader,
+        *, 
+        **kwargs,
+        # removed batch_size, it is not required that it will accept kwargs
+    ) -> np.ndarray:  # added standard output (I believe we actually expect tensors in the code, but would like to be consistent)
+        pass
+
+    def get_text_embeddings( # any reason for this?
+        self,
+        texts: list[str],
+        *, 
+        **kwargs, 
+    ) -> np.ndarray:
+        pass
+
+    def get_fused_embeddings( # hmm what if I have a document with images at specific positions?
```

Comment:
Sort of like the way it's handled in vlm2vec, where the text would indicate where the image should be interleaved (via `<|image_1|>`)?

Reply:
Yep exactly